core/store/src/trie/mem/flexible_data/children.rs

```diff
@@ -0,0 +1,99 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+
+use super::encoding::BorshFixedSize;
+use super::FlexibleDataHeader;
+use crate::trie::mem::arena::{ArenaSlice, ArenaSliceMut};
+use crate::trie::mem::node::{MemTrieNodeId, MemTrieNodePtr};
+use crate::trie::Children;
+
+/// Flexibly-sized data header for a variable-sized list of children trie nodes.
+/// The header contains a 16-bit mask of which children are present, and the
+/// flexible part is one pointer for each present child.
+#[derive(Clone, Copy, BorshSerialize, BorshDeserialize)]
+pub struct EncodedChildrenHeader {
+    mask: u16,
+}
+
+impl BorshFixedSize for EncodedChildrenHeader {
+    const SERIALIZED_SIZE: usize = std::mem::size_of::<u16>();
+}
+
+impl FlexibleDataHeader for EncodedChildrenHeader {
+    type InputData = Vec<Option<MemTrieNodeId>>;
+    type View<'a> = ChildrenView<'a>;
+
+    fn from_input(children: &Vec<Option<MemTrieNodeId>>) -> EncodedChildrenHeader {
+        let mut mask = 0u16;
+        for i in 0..16 {
+            if children[i].is_some() {
```

Comment:
Take `children: &[Option<MemTrieNodeId>; 16]`, if the vec must be of size 16 anyway?

Reply:
Oh yeah I need to fix this. In the other implementation you reviewed before, I couldn't do this because the node didn't implement Copy. But now I can.